pkg/routeagent/controllers/route/route.go

```diff
@@ -243,10 +476,19 @@ func (r *Controller) enqueueEndpoint(obj interface{}) {
 		utilruntime.HandleError(err)
 		return
 	}
-	klog.V(4).Infof("Enqueueing endpoint for route controller %v", obj)
 	r.endpointWorkqueue.AddRateLimited(key)
 }
 
+func (r *Controller) enqueuePod(obj interface{}) {
+	klog.V(6).Infof("Enqueueing pod for route controller %v", obj)
+	pod := obj.(*k8sv1.Pod)
+
+	// Add the POD event to the workqueue only if the sm-route-agent podIP does not exist in the local cache.
+	if !r.remoteVTEPs.Contains(pod.Status.HostIP) {
+		r.podWorkqueue.AddRateLimited(obj)
+	}
+}
+
```

Comment:
The `vxlanDevice` is created via an _Endpoint_ add - should `handleRemovedEndpoint` cleanup the `vxlanDevice`?

Reply:
Good point. I'll update the code. Thanks.